go/store/nbs/table_reader.go

```diff
@@ -75,14 +74,31 @@ func (cmp CompressedChunk) ToChunk() (chunks.Chunk, error) {
 	return chunks.NewChunk(data), nil
 }
 
+func ChunkToCompressedChunk(chunk chunks.Chunk) CompressedChunk {
+	compressed := snappy.Encode(nil, chunk.Data())
+	length := len(compressed)
+	compressed = append(compressed, []byte{0, 0, 0, 0}...)
+	binary.BigEndian.PutUint32(compressed[length:], crc(compressed[:length]))
+	return CompressedChunk{H: chunk.Hash(), FullCompressedChunk: compressed, CompressedData: compressed[:length]}
+}
+
 // Hash returns the hash of the data
 func (cmp CompressedChunk) Hash() hash.Hash {
 	return cmp.H
 }
 
 // IsEmpty returns true if the chunk contains no data.
 func (cmp CompressedChunk) IsEmpty() bool {
-	return len(cmp.CompressedData) == 0
+	return len(cmp.CompressedData) == 1 && cmp.CompressedData[0] == 0
+}
+
+var EmptyCompressedChunk CompressedChunk
+
+func init() {
+	EmptyCompressedChunk = ChunkToCompressedChunk(chunks.EmptyChunk)
+	if !EmptyCompressedChunk.IsEmpty() {
+		panic("EmptyCompressedChunk should be empty...")
+	}
```

Comment:
@bheni Should this be here? Or in a unit test?

Reply:
I guess a unit test.  Better for Jenkins to fail it rather than for it to panic at startup.

Reply:
Move it!